state-chain/traits/src/lib.rs

```diff
@@ -952,3 +954,16 @@ pub trait AssetConverter {
 pub trait IngressEgressFeeApi<C: Chain> {
 	fn accrue_withheld_fee(asset: C::ChainAsset, amount: C::ChainAmount);
 }
+
+pub trait Refunding<C: Chain> {
+	fn record_gas_fees(account_id: C::ChainAccount, asset: C::ChainAsset, amount: C::ChainAmount);
+	fn with_held_transaction_fees(asset: C::ChainAsset, amount: C::ChainAmount);
```

Comment:
It might be simpler to define the trait in terms of `AssetAmount` and `cf_primitives::Asset` instead of making it chain-generic. Then we can just implement it on the pallet, and there's no need to define a macro for all the trait defs. We can just use `into()` where we call the methods.